### PR TITLE
[fix](nereids)build agg for random distributed agg table in bindRelation phase

### PR DESCRIPTION
**Pull request by starocean999**
it's better to build agg for random distributed agg table in bindRelation phase instead of in BuildAggForRandomDistributedTable RULE

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Analyzer.java [hunk 1]
```diff
@@ -163,8 +162,6 @@ private static List<RewriteJob> buildAnalyzerJobs(Optional<CustomTableResolver>
             topDown(new EliminateGroupByConstant()),
 
             topDown(new SimplifyAggGroupBy()),
-            // run BuildAggForRandomDistributedTable before NormalizeAggregate in order to optimize the agg plan
-            topDown(new BuildAggForRandomDistributedTable()),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Analyzer.java:167]: remove this file BuildAggForRandomDistributedTable and rule type for it

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java [hunk 2]
```diff
@@ -215,7 +236,103 @@ private LogicalPlan makeOlapScan(TableIf table, UnboundRelation unboundRelation,
                     unboundRelation.getTableSample());
             }
         }
-        return checkAndAddDeleteSignFilter(scan, ConnectContext.get(), (OlapTable) table);
+        if (needGenerateLogicalAggForRandomDistAggTable(scan)) {
+            return preAggForRandomDistribution(scan);
+        } else {
+            return checkAndAddDeleteSignFilter(scan, ConnectContext.get(), (OlapTable) table);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:242]: add comments to explain why these two function is mutually

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java [hunk 3]
```diff
@@ -215,7 +236,103 @@ private LogicalPlan makeOlapScan(TableIf table, UnboundRelation unboundRelation,
...
+        }
+    }
+
+    private boolean needGenerateLogicalAggForRandomDistAggTable(LogicalOlapScan olapScan) {
+        if (ConnectContext.get() != null && ConnectContext.get().getState() != null
+                && ConnectContext.get().getState().isQuery()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:248]: add comments to explain why only query

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Review by 924060929 — COMMENTED** [review]
This change maybe conflict to CheckPolicy because it only support filter-relation or relation, please add row policy / data masking test case

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**